R/get-osmdata.R

```diff
@@ -745,6 +746,46 @@ xml_to_df <- function (doc, stringsAsFactors = FALSE) {
 }
 
 
+xml_to_df_cpp <- function (doc, stringsAsFactors = FALSE) {
+
+    res <- rcpp_osmdata_df (paste0 (doc))
+    if (nrow (res$points_kv) > 0L) {
+        res$points_kv$osm_type <- "node"
+    }
+    if (nrow (res$ways_kv) > 0L) {
+        res$ways_kv$osm_type <- "way"
+    }
+    if (nrow (res$rels_kv) > 0L) {
+        res$rels_kv$osm_type <- "relation"
+    }
+
+    nms <- sort (unique (unlist (lapply (res, names))))
+    nms1 <- c ("osm_type", "osm_id", "name")
+    nms <- c (nms1, setdiff (nms, nms1))
+
+    df <- lapply (res, function (i) {
+        out <- data.frame (
+            matrix (nrow = nrow (i), ncol = length (nms)),
+            stringsAsFactors = stringsAsFactors
+        )
+        names (out) <- nms
+        out [, names (i)] <- i
+        return (out)
+    })
+    df <- do.call (rbind, df)
+
+    if (nrow (df) == 0) {
+        df <- data.frame (
+            osm_type = character (),
+            osm_id = character (),
+            stringsAsFactors = stringsAsFactors
+        )
+    }
+
```

Comment:
```suggestion

    rownames (df) <- NULL

```
Remove uninformative row names